### PR TITLE
Indicate what root privileges are needed for in kickstart.sh.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

Instead of just blindly prompting the user for root privileges when they are needed in the kickstart script, print a message about _why_ we need root privileges. This improves transparency for users about why they are being prompted.

##### Test Plan

Run the kickstart script from this PR as a non-root user. You should see messages about root privileges being required for specific commands as those commands are run.

### CODE REVIEW

File: packaging/installer/kickstart.sh [hunk 1]
```diff
@@ -1217,14 +1227,14 @@ set_auto_updates() {
     # This first case is for catching using a new kickstart script with an old build. It can be safely removed after v1.34.0 is released.
     elif ! grep -q '\-\-enable-auto-updates' ${updater}; then
       echo
-    elif ! ${ROOTCMD} ${updater} --enable-auto-updates "${NETDATA_AUTO_UPDATE_TYPE}"; then
+    elif ! run_as_root ${updater} --enable-auto-updates "${NETDATA_AUTO_UPDATE_TYPE}"; then
       warning "Failed to enable auto updates. Netdata will still work, but you will need to update manually."
```
Comment (thiagoftsm) [inline packaging/installer/kickstart.sh:1231]: Hello @Ferroin ,

`shellcheck` is reporting issues in the modified lines, are you going to address these issues?

best regards!

**Comment by thiagoftsm** [comment]
Hello @Ferroin ,

I was testing your PR on FreeBSD as a normal user, but I accessed the normal user running `su - ` on a new environment and the first question I received was:

```sh
[vagrant@freebsd13 ~/netdata]$ ./packaging/installer/kickstart.sh 

 --- Using /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- No existing installations of netdata found, assuming this is a fresh install. --- 
 --- Attempting to install by building locally... --- 
[/tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM]$ curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh 
 OK 

 --- Running downloaded script to install required build dependencies... --- 
Root privileges required to run /bin/bashRoot privileges required to run /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.shRoot privileges required to run netdata[/tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM]$ sudo /bin/bash /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.sh netdata 
 > FreeBSD Version: 13.1-RELEASE-p3 ...
 > Checking for gmake ...
gmake is required to build on FreeBSD and is not installed. Shall I install it? [y/n] 
```

Considering this output, should we not required root permission before to reach this step?

Best regards!

**Comment by Ferroin** [comment]
> Hello @Ferroin ,
> 
> I was testing your PR on FreeBSD as a normal user, but I accessed the normal user running `su - ` on a new environment and the first question I received was:
> 
> ```shell
> [vagrant@freebsd13 ~/netdata]$ ./packaging/installer/kickstart.sh 
> 
>  --- Using /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM as a temporary directory. --- 
>  --- Checking for existing installations of Netdata... --- 
>  --- No existing installations of netdata found, assuming this is a fresh install. --- 
>  --- Attempting to install by building locally... --- 
> [/tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM]$ curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh 
>  OK 
> 
>  --- Running downloaded script to install required build dependencies... --- 
> Root privileges required to run /bin/bashRoot privileges required to run /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.shRoot privileges required to run netdata[/tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM]$ sudo /bin/bash /tmp/netdata-kickstart-XXXXXXXXXX.0Q7kEGWM/install-required-packages.sh netdata 
>  > FreeBSD Version: 13.1-RELEASE-p3 ...
>  > Checking for gmake ...
> gmake is required to build on FreeBSD and is not installed. Shall I install it? [y/n] 
> ```
> 
> Considering this output, should we not required root permission before to reach this step?

No, this actually is the first time we need root permissions in the install script, though I need to update the message formatting to properly reflect what’s actually being run.

**Review by thiagoftsm — APPROVED** [review]
Everything worked as expected:

```sh
vagrant@ubuntu2204:/home/thiago/Netdata/netdata$ ./packaging/installer/kickstart.sh 

 --- Using /tmp/netdata-kickstart-IUltx2afXU as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- No existing installations of netdata found, assuming this is a fresh install. --- 
 --- Attempting to install using native packages... --- 
 --- Checking for availability of repository configuration package. --- 
[/tmp/netdata-kickstart-IUltx2afXU]$ curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-IUltx2afXU/netdata-repo-edge_2-1+ubuntu22.04_all.deb https://repo.netdata.cloud/repos/repoconfig/ubuntu/jammy/netdata-repo-edge_2-1+ubuntu22.04_all.deb 
 OK 

Root privileges required to run env
Root privileges required to run apt-get
Root privileges required to run update
[/tmp/netdata-kickstart-IUltx2afXU]$ sudo env apt-get update 
Hit:1 https://mirrors.edge.kernel.org/ubuntu jammy InRelease
Get:2 https://mirrors.edge.kernel.org/ubuntu jammy-updates InRelease [114 kB]
Get:3 https://mirrors.edge.kernel.org/ubuntu jammy-backports InRelease [99.8 kB]
Get:4 https://mirrors.edge.kernel.org/ubuntu jammy-security InRelease [110 kB]
Get:5 https://mirrors.edge.kernel.org/ubuntu jammy-updates/main amd64 Packages [834 kB]                                                                                                                                                    
Get:6 https://mirrors.edge.kernel.org/ubuntu jammy-updates/main Translation-en [184 kB]                                                                                                                                                    
Get:7 https://mirrors.edge.kernel.org/ubuntu jammy-updates/main amd64 c-n-f Metadata [12.2 kB]                                                                                                                                             
Get:8 https://mirrors.edge.kernel.org/ubuntu jammy-updates/universe amd64 Packages [790 kB]                                                                                                                                                
Get:9 https://mirrors.edge.kernel.org/ubuntu jammy-updates/universe Translation-en [138 kB]                                                                                                                                                
Get:10 https://mirrors.edge.kernel.org/ubuntu jammy-updates/universe amd64 c-n-f Metadata [14.8 kB]                                                                                                                                        
Get:11 https://mirrors.edge.kernel.org/ubuntu jammy-security/main amd64 Packages [588 kB]                                                                                                                                                  
Get:12 https://mirrors.edge.kernel.org/ubuntu jammy-security/main amd64 c-n-f Metadata [7,684 B]                                                                                                                                           
Get:13 https://mirrors.edge.kernel.org/ubuntu jammy-security/universe amd64 Packages [632 kB]                                                                                                                                              
Get:14 https://mirrors.edge.kernel.org/ubuntu jammy-security/universe Translation-en [85.3 kB]                                                                                                                                             
Get:15 https://mirrors.edge.kernel.org/ubuntu jammy-security/universe amd64 c-n-f Metadata [11.1 kB]                                                                                                                                       
Fetched 3,621 kB in 36s (100 kB/s)                                                                                                                                                                                                         
Reading package lists... Done
 OK 

Root privileges required to run env
Root privileges required to run apt-get
Root privileges required to run install
Root privileges required to run /tmp/netdata-kickstart-IUltx2afXU/netdata-repo-edge_2-1+ubuntu22.04_all.deb
```

LGTM!

**Review by underhood — DISMISSED** [review]

**Comment by Ferroin** [comment]
Rebased to pick up CI changes.

Also fixed an issue with the message formatting.

**Comment by underhood** [comment]
[f9616a1](https://github.com/netdata/netdata/pull/14314/commits/f9616a12c839b09eb97734bae8b96e85e8a2e073) LOL how did we not notice that in the output
```
Root privileges required to run apt-get
Root privileges required to run install
...
```

**Review by thiagoftsm — APPROVED** [review]

**Review by underhood — APPROVED** [review]

**Pull request merged**